ports/nanobind/vcpkg.json

```
@@ -0,0 +1,14 @@
+{
+  "name": "nanobind",
+  "version-date": "2023-12-16",
```

Note: This should be changed to
```suggestion
  "version-semver": "1.9.0",
```
when the next version (likely `1.9.0`) is released.

I have chosen `version-semver` based on the text in nanobind's [changelog file](https://github.com/wjakob/nanobind/blob/16515dbe198d9406fad046d7fc6e7c4ca5e2c5a6/docs/changelog.rst)

---

N/A. I've decided to backport the CMake patches required for packaging.